### DISCUSSION TITLE
Research Stations drops items on the ground

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#13922] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

The research station tends to drop items on the ground for 2 reasons:

1. The bottom row / products has containable restrictions for no reason. The 1st row needs containable tags because otherwise it could be used as a regular deconstructor, but the 2nd row doesnt.

2. Only 1 slot in 2nd row but 2 slots in 1st row means sometimes items drop on the floor. Doing an "invalid recipe" will drop one of the items on the floor
    
Every time a mod wants to add research station interactions, they have to override, resulting in conflicts.

**Change the xml from**

```xml
    <ItemContainer capacity="2" maxstacksize="1" canbeselected="true" hideitems="true" hudpos="0.5, 0.4" slotsperrow="3" uilabel="" allowuioverlap="true">
      <Containable items="geneticmaterial,stabilozine,unidentifiedgeneticmaterial" />
    </ItemContainer>
    <ItemContainer capacity="1" maxstacksize="1" canbeselected="true" hideitems="true" hudpos="0.5, 0.8" slotsperrow="5" uilabel="" allowuioverlap="true">
      <Containable items="geneticmaterial,stabilozine,unidentifiedgeneticmaterial" />
    </ItemContainer>
```

To
* Added "researchitem" tag to the top row, general purpose tag that can be used by mods for items that arent genetics
* Removed containables from bottom row

```xml
    <ItemContainer capacity="2" maxstacksize="1" canbeselected="true" hideitems="true" hudpos="0.5, 0.4" slotsperrow="3" uilabel="" allowuioverlap="true">
      <Containable items="geneticmaterial,stabilozine,unidentifiedgeneticmaterial,researchitem" />
    </ItemContainer>
    <ItemContainer capacity="2" maxstacksize="1" canbeselected="true" hideitems="true" hudpos="0.5, 0.8" slotsperrow="5" uilabel="" allowuioverlap="true">
    </ItemContainer>
```

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/1e26b4cd-d2cf-4c38-a3cc-03453e81dc7c)

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

**Comment by UnhappyWithMyUsername** [comment]
I highly recommend to add the following as a default value for `<Deconstruct>` components, if nothing is configured by the modder:
`requireddeconstructor="deconstructor"`

This way if a modder for example just puts `<Deconstruct time="5" />`, it will automatically only become valid for the regular deconstructor and does not affect the item when put into a research station or other custom deconstructors unless it was specifically configured in the `<Deconstruct>` component. This would also solve the issue of any vanilla item getting deconstructed if put into a research station if that research station allows any item input.

**Comment by AzureNocte** [comment]
But why change the well-functioning vanilla research stations? You could make a modded version of it.

**Reply by ubertpendragon** [reply]
Modding conflicts, if many mods have to override research station or items then you get incompatibilities

I'm suggesting to make research station more compatible in vanilla

**Comment by Regalis11** [comment]
I'm not entirely sure I understand what you're intending to do here. Do you mean that you would like to be able to refine/process things other than genetic materials in the research station, but without overriding the research station? To me that doesn't seem ideal, considering that e.g. the texts on the research station talk about genetic materials.

**Reply by ubertpendragon** [reply]
Basically:

1. Mods want to add functionalities to the research station, to expand on it
2. Base game has strict container restrictions
3. Mods override the station each on their own
4. Modding conflicts


I'm suggesting to add a universal `researchitem` tag in containables that any mod can use to prevent conflicts (just like `handheldmagazine` was added on unstable, very helpful), as well as fixing the station dropping or not being able to contain items due to containable restrictions

This wouldn't change anything for vanilla, it would just help mods

**Reply by Regalis11** [reply]
I get that part, I'm just wondering what kind of expansions you're thinking of. Something else than genes? Won't that be confusing to players, considering the research station UI talks about genes?

**Reply by UnhappyWithMyUsername** [reply]
The research stations actually don't reference genetic material but rather talk about "Alien Material" which would allow adding different alien things to research and still have the text be valid. I make use of that in "Artifacts and Ruins Enhanced" to research artifacts to actually figure out what you're dealing with and to unlock new technologies to play around with.

```
<researchstation.research.infotext>Alien material inserted. Use caution.</researchstation.research.infotext>
<researchstation.combine.missingitem>Alien material inserted. Accelerant required ([itemname]).</researchstation.combine.missingitem>
```

**Reply by ubertpendragon** [reply]
> what kind of expansions you're thinking of. Something else than genes? 

Well since we're spitballing ideas: 

- Recycling genes into temporary injections (potion technology acquired)

https://github.com/FakeFishGames/Barotrauma/assets/73229309/e0c251ab-76df-4294-95f0-20dc8e323be0

- The equivalent of scrap for plants (we had to use a workaround for plants to not fall off the table)

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/2b35c0ad-1b0d-4b42-9538-a0486ad4a7db)

- Talents that could use research station such as:

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/91ee6a07-d9f6-436e-8ecc-4d9a6f16d5d3)

- Partially recharging alien batteries with fulgurium/alien circuits


**You could say, why not use a fabricator/deconstructor instead?** 

1. Because flavor is important
2. This sets research outposts apart. In Dynamic Europa we removed the research station from the engineering module,
3. Research station feels like abandoned content. Only used for genes and doesn't even have a broken sprite

I'm tempted to open a PR

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/commit/699912c0071c81f8d0a1929d1b7ff16d46a57a85 by adding the "researchmaterial" tag to containables in the input slots, and "researchmaterial", "smallitem" and "mediumitem" to the output slots.

**Reply by ubertpendragon** [reply]
pog
